modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -106,5 +111,35 @@ object ConstructionService {
       } else {
         parseUnsignedTransaction(hex)
       }
+
+    def combineTransaction(hex: Hex, signatures: NonEmptyList[RosettaSignature]): EitherT[F, ConstructionError, Hex] =
+      EitherT
+        .fromEither(KryoSerializer[F].deserialize[Transaction](hex.toBytes))
+        .leftMap(_ => MalformedTransaction)
+        .flatMap { transaction =>
+          convertSignatures(signatures).flatMap { signatures =>
+            EitherT
+              .fromEither(
+                KryoSerializer[F]
+                  .serialize(Signed[Transaction](transaction, NonEmptySet(signatures.head, SortedSet(signatures.tail: _*))))
+              )
+              .leftMap(_ => MalformedTransaction.asInstanceOf[ConstructionError])
+              .map(Hex.fromBytes(_))
+          }
+        }
+
+    def convertSignatures(
+      signatures: NonEmptyList[RosettaSignature]
+    ): EitherT[F, ConstructionError, NonEmptyList[SignatureProof]] =
+      EitherT {
+        signatures.traverse { signature =>
+          signature.publicKey.hexBytes.toPublicKeyByEC.map { pk =>
+            signature.signatureType match {
+              case SignatureType.ECDSA => SignatureProof(pk.toId, Signature(signature.hexBytes)).asRight[ConstructionError]
+              case _                   => UnsupportedOperation.asInstanceOf[ConstructionError].asLeft[SignatureProof]
```

Comment:
FYI - I'm pretty sure this case would never happen, as the request would be rejected on the very first operation (trying to decode the request), as the ECDSA signature is the only one that is supported, so there is no possibility you can get any other type here (hence, match should have also a single element probably)

Reply:
The logic has been reworked and this function is no more.